eng/validation/update-channel.ps1

```diff
@@ -14,56 +15,72 @@ function Get-Headers([string]$accept, [string]$barToken) {
 $getAssetsApiEndpoint = "$maestroEndpoint/api/assets?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=$apiVersion"
 $headers = Get-Headers 'text/plain' $barToken
 
-# Get the Microsoft.DotNet.Arcade.Sdk with the version $arcadeSdkVersion so we can get the id of the build
-$assets = Invoke-WebRequest -Uri $getAssetsApiEndpoint -Headers $headers | ConvertFrom-Json
+try {
+    # Get the Microsoft.DotNet.Arcade.Sdk with the version $arcadeSdkVersion so we can get the id of the build
+    $assets = Invoke-WebRequest -Uri $getAssetsApiEndpoint -Headers $headers | ConvertFrom-Json
 
-if (!$assets) {
-    Write-Host "Asset '$arcadeSdkPackageName' with version $arcadeSdkVersion was not found"
-    exit 1
-}
+    if (!$assets) {
+        Write-Host "Asset '$arcadeSdkPackageName' with version $arcadeSdkVersion was not found"
+        exit 1
+    }
 
-if ($assets.Count -ne 1) {
-    Write-Host "More than 1 asset matched the version '$arcadeSdkVersion' of Microsoft.DotNet.Arcade.Sdk. This is not normal. Stopping execution..."
-    exit 1
-}
+    if ($assets.Count -ne 1) {
+        Write-Host "More than 1 asset matched the version '$arcadeSdkVersion' of Microsoft.DotNet.Arcade.Sdk. This is not normal. Stopping execution..."
+        exit 1
+    }
 
-$buildId = $assets[0].'buildId'
+    $buildId = $assets[0].'buildId'
 
-# When we merge an internal change in arcade-validation an official build will be triggered. This won't come from Maestro. In this case the SDK version
-# was already validated and added to $targetChannelName. So we need to check if the build is already in $targetChannelName so we don't fail trying to
-# update the channel each time
-$getBuildApiEndpoint = "$maestroEndpoint/api/builds/$buildId/?api-version=$apiVersion"
-$build = Invoke-WebRequest -Uri $getBuildApiEndpoint -Headers $headers | ConvertFrom-Json
-$buildInChannel = $($build.channels | Where-Object -Property "name" -Value "${targetChannelName}" -EQ)
+    . $PSScriptRoot\..\common\darc-init.ps1
 
-if ($buildInChannel) {
-    Write-Host "Build '$buildId' is already in channel '$targetChannelName'. This is most likely an arcade-validation internal build"
-    exit 0
-}
+    $DarcOutput = darc add-build-to-channel --id $buildId --channel "$targetChannelName"
+    
+    if ($LastExitCode -ne 0) {
+        Write-Host "Problems using Darc to promote build ${buildId} to channel ${targetChannelName}. Stopping execution..."
+        Write-Host $DarcOutput
+        exit 1
+	}
 
-$getChannelsEndpoint = "$maestroEndpoint/api/channels?api-version=2018-07-16"
-$channels = Invoke-WebRequest -Uri $getChannelsEndpoint -Headers $headers | ConvertFrom-Json
-$matchedChannel = $($channels | Where-Object -Property "name" -Value "${targetChannelName}" -EQ | Select-Object -Property id)
+    if ($DarcOutput -match "has already been assigned to") 
+    {
+        Write-Host "Build has already been assigned to channel '${targetChannelName}'."
```

Comment:
The old message had a little more information. 

`Write-Host "Build '$buildId' is already in channel '$targetChannelName'. This is most likely an arcade-validation internal build`